kubernetes_platform/python/kfp/kubernetes/toleration.py

```diff
@@ -82,21 +81,46 @@ def add_toleration(
 
 
 def add_toleration_json(task: PipelineTask,
-                        toleration_json: Union[pipeline_channel.PipelineParameterChannel, dict]
+                        toleration_json: Union[pipeline_channel.PipelineParameterChannel, list, dict]
                         ):
-    """Add a Pod Toleration in the form of a JSON to a task.
+    """Add a Pod Toleration in the form of a Pipeline Input JSON to a task.
 
     Args:
         task:
             Pipeline task.
         toleration_json:
-            a toleration provided as dict or input parameter. Takes
-            precedence over other key, operator, value, effect,
-            and toleration_seconds.
+            a toleration that is a pipeline input parameter.
+            The input parameter must be of type dict or list.
+
+            If it is a dict, it must be a single toleration object.
+            For example a pipeline input parameter in this case could be::
+                {
+                  "key": "key1",
+                  "operator": "Equal",
+                  "value": "value1",
+                  "effect": "NoSchedule"
+                }
 
+            If it is a list, it must be list of toleration objects.+
+            For example a pipeline input parameter in this case could be:
+                [
+                    {
+                      "key": "key1",
+                      "operator": "Equal",
+                      "value": "value1",
+                      "effect": "NoSchedule"
+                    },
+                    {
+                      "key": "key2",
+                      "operator": "Exists",
+                      "effect": "NoExecute"
+                    }
+                ]
     Returns:
         Task object with added toleration.
     """
+    if not isinstance(toleration_json, pipeline_channel.PipelineParameterChannel):
+        raise TypeError("toleration_json must be a Pipeline Input Parameter.")
```

Comment:
Why is this required? It seems reasonable to allow static values.

Reply:
this one we can discuss but currently the support is only for pipelienparameter channels and the check is protecting the user from making a mistake only recognizable during runtime (without this), if you are provided hardcoded values  then `add_toleration()` is largely sufficient for most needs even list of dicts, adding support for static list/dicts is out of scope, but I'd be happy to add this in a follow up if we want to go that route

Reply:
added support for static list/dicts. One caveat here is that the pipeline spec yaml only will utilize the toleration_json for input parameter under the hood, since this isn't used for add_toleration.